### ISSUE TITLE
Support grouping pipelineRuns

### ISSUE THREAD

**Issue by drriguz** [tektoncd/dashboard#2335]
# Is your feature request related to a problem ?
My understanding of tekton is, the pipelines is supposed to be reused by multiple projects(unlike Jenkins or travis ci, each project/repository owns a pipeline). The problem is that there's currently no way to view the CI status in projects level, users are not able to find there projects easily. 

In Jenkins, there's monitor that shows the latest status of the project:
![image](https://user-images.githubusercontent.com/6427851/162141860-e655218b-9c5b-4d9d-bd74-bfa1ae33ef2a.png)

# Describe the solution you'd like

Since there's actually no concept of "project" in tekton, I don't think we should create UI for projects. Instead, what I'm thinking is that, allow user to group the pipelineRuns by tags(and by pipelines). It could be fully customized by the user to control the tags, in our projects, we create a "app" tag for identifing the project:

![image](https://user-images.githubusercontent.com/6427851/162142484-5b3e8d1c-c80c-4502-84e2-438443bbfa39.png)

So, if we could show the pipelineRuns by grouping using "app" tag, then it equivlent to a projects dashboard.

Also, since the UI is grouped, and shows only the latest pipelineRuns matches the tag, there should be significant performance improvment, which could also be considered to solve https://github.com/tektoncd/dashboard/issues/2324

**Comment by AlanGreene** [comment]
Thanks for the feature request @drriguz. As you mentioned this would need to be customised by the user and different users may have different needs or expectations for how runs should be grouped or displayed. This will need some design exploration but is something we may consider in the future.

**Comment by drriguz** [comment]
Thanks for your detailed explaination, that totally make sense. So as you mentioned, currently the dashboard is highly coupled with the Kubernates API, will it be considered to add a real backend(I guess currently the backend is just a proxy of Kubernates APIs) for the dashboard? Furtuer more, if there's backend the the APIs could be exposed for the user, and uses might have a chance to build a customized UI by themselves, while benefit from the features provided by dashboard(eg. realtime updates).

One possible approach might be integrating with tekton results, and the tekton result team is already working on that, https://github.com/tektoncd/results/issues/82. Could you also share if there's any plan/updates for the integration?

**Comment by AlanGreene** [comment]
The Dashboard originally provided its own custom APIs but we switched to proxying the Kubernetes API for a number of reasons including flexibility, reducing the maintenance overhead, and the ability to take advantage of Kubernetes RBAC for auth.

Realistically I think we're unlikely to switch back again anytime soon. At a minimum this would mean having to implement auth ourselves, and the back end would still be subject to the limitations of the Kubernetes API as that's still the mechanism through which we'd receive the resources. Caching is also problematic as we would need to mindful of auth issues, ensuring we invalidate caches affected by RBAC changes. Doing this would also require the Dashboard ServiceAccount to be able to monitor RBAC changes (roles, bindings, etc.) which it does not currently do, and which it really should not require.

Tekton Results integration is still on the roadmap, but the Results project currently supports only a subset of the Tekton resources. The issue you referenced is the correct one to track for updates.

The Dashboard used to provide an extension mechanism which allowed users to add custom UI, for example the experimental Webhooks extension which is no longer maintained / supported and was removed about a year ago, see https://github.com/tektoncd/experimental/issues/750. Our current extension mechanism is quite basic and allows for the addition of other Kubernetes resources to the UI, see the docs for more https://github.com/tektoncd/dashboard/blob/main/docs/extensions.md

I'll leave this issue open for further discussion and we may revisit the feature request in future.
